trunner/harness/unity.py

```diff
@@ -27,14 +27,15 @@ def unity_harness(dut: Dut, ctx: TestContext, result: TestResult) -> Optional[Te
             if parsed["status"] in ["FAIL", "IGNORE"]:
                 last_assertion = parsed
         elif idx in (1, 2):
+            msg = ""
             if last_assertion:
-                parsed["msg"] = last_assertion["msg"]
+                msg = last_assertion["msg"]
                 last_assertion = {}
 
             status = Status.from_str(parsed["status"])
             subname = f"{parsed['group']}.{parsed['name']}"
             if "path" in parsed and "line" in parsed:
-                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + parsed["msg"]
+                parsed["msg"] = f"[{parsed['path']}:{parsed['line']}] " + msg
             result.add_subresult(subname, status, parsed.get("msg", ""))
```

Comment:
don't need to use `.get` when we have declared msg earlier as empty, also can put everything into a formatted string
```suggestion
                msg = f"[{parsed['path']}:{parsed['line']}] {msg}"
            result.add_subresult(subname, status, msg)
```